### PR TITLE
[WINDOWS] Compiler options for non x86 targets

### PR DESCRIPTION
**Pull request by krishnaraj36**
Windows compiler options for targets

### CODE REVIEW

**Comment by krishnaraj36** [comment]
@srkreddy1238  Can you please look into this PR

File: python/tvm/contrib/cc.py [hunk 1]
```diff
@@ -373,7 +373,11 @@ def _linux_compile(
 
 def _windows_compile(output, objects, options, cwd=None, ccache_env=None):
     cmd = ["clang"]
+    compiler = os.getenv("TVM_WIN_CC", default="clang")
+    win_target = os.getenv("TVM_WIN_TARGET", default="x86_64")
+    cmd = [compiler]
```
Comment (cbalint13) [inline python/tvm/contrib/cc.py:378]: This overwrite initial ```cmd = [clang]``` initialization, in a counterintuitive logic, could we condition the new assignment here to ```TVM_WIN_TARGET``` presence ?

File: python/tvm/contrib/cc.py [hunk 2]
```diff
@@ -373,7 +373,11 @@ def _linux_compile(
...
     cmd += ["-O2"]
+    cmd += ["--target=" + win_target]
```
Comment (cbalint13) [inline python/tvm/contrib/cc.py:380]: Let's also condition these too within the above comment.

**Review by cbalint13 — CHANGES_REQUESTED** [review]
Thank you @krishnaraj36 for the contribution !

Looks good to me, lets have the assignment logic and the CI lint issue fixed in order to get this merged.

File: python/tvm/contrib/cc.py (hunk 1 above)
Comment (cbalint13) [reply python/tvm/contrib/cc.py:378]: Also lets fix ```cwd=None``` typo here, perhaps we could have ```cmd=[clang]``` (if you would like better) and then just override with windows env related conditional.

Comment (krishnaraj36) [reply python/tvm/contrib/cc.py:378]: @cbalint13   make sense, we don't require this assignment as we are consuming from environment variables.

**Review by srkreddy1238 — APPROVED** [review]
LGTM, let the CI pass.

File: python/tvm/contrib/cc.py (hunk 1 above)
Comment (cbalint13) [reply python/tvm/contrib/cc.py:377]: This is fine too (removing), I am sure the original would worked but its logic was strange. 

Thank you !

 Apollogise for confusion of ```cwd``` vs. ```cmd``` function parameter, it is not the case to involve that too.

**Review by cbalint13 — APPROVED** [review]

File: python/tvm/contrib/cc.py [hunk 3]
```diff
@@ -173,8 +173,8 @@ def create_executable(output, objects, options=None, cc=None, cwd=None, ccache_e
 
     if _is_linux_like():
         _linux_compile(output, objects, options, cc, cwd, ccache_env)
-    elif _is_windows_like():
-        _windows_compile(output, objects, options, cwd, ccache_env)
+    elif _is__like():
+        __compile(output, objects, options, cwd, ccache_env)
```
Comment (cbalint13) [inline python/tvm/contrib/cc.py:177]: Does changes to ```_wimdows_compile()``` still make sense since now it is switched to ```_compile()```  ?

**Pull request merged**